File: Python3/342. Power of Four.py
class Solution:
    
    def __init__(self):
        self.nums = [1] * 16
        for i in range(1, 16):
            self.nums[i] = 4 * self.nums[i-1]
        
    def isPowerOfFour(self, n: int) -> bool:
        return n in self.nums

File: Java/342. Power of Four.java
class Solution {
    private Power4 p = new Power4();
    public boolean isPowerOfFour(int n) {
        return p.nums.contains(n);
        
    }
}

class Power4 {
    public List<Integer> nums = new ArrayList();
    
    Power4() {
        int last = 1;
        nums.add(last);
        
        for (int i=1; i<16; i++) {
            last *= 4;
            nums.add(last);
        }
    }
}